JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/ProjectBean.java

```diff
@@ -1,211 +1,282 @@
+        if (!hash.equals("")) {
+            return createCopy(hash);
 
-        if(hash == null) {
-            throw new IllegalArgumentException("Error: hash argument cannot be null");
         }
-        //New project
-        if(hash.equals("")) {
-
-            User user = userBean.findUser(creator);
-            if(user == null) {
-                throw new IllegalArgumentException("Error: the creator of the project must exist");
-            }
-
-            Access access;
-            try {
-                access = (Access) em.createQuery("SELECT a FROM Access a WHERE a.accessName =:access")
-                        .setParameter("access", Access.READ_AND_WRITE)
-                        .getSingleResult();
-            } catch (NoResultException e){
-                throw new RuntimeException("Project can not be created");
-            }
-
-            Hash newHash = new Hash();
-            Project project = new Project();
-            UserGroup userGroup = new UserGroup();
-            Group group = new Group();
-            List<UserGroup> usergroup = new LinkedList<>();
-            usergroup.add(userGroup);
-            List<Project> projects = new LinkedList<>();
-            projects.add(project);
-
-            newHash.setHash("123456789");//!TODO hash generator
-            em.persist(newHash);
-
-            group.setGroupName(DEFAULT_GROUP_NAME);
-            group.setMembers(usergroup);
-            group.setProjects(projects);
-            em.persist(group);
-
-            userGroup.setGroup(group);
-            userGroup.setMember(user);
-            userGroup.setAccess(access);
-            userGroup.setUserId(user.getUserId());
-            userGroup.setGroupId(group.getGroupId());
-            em.persist(userGroup);
-
-            project.setGroup(group);
-            project.setHash(newHash);
-            project.setProjectName(projectName);
-            em.persist(project);
-
-            return newHash.getHash();
+        Group group;
+        Project project = new Project();
+        Hash hashes = new Hash();
+
+        //get group, corresponding to this id
+        group = getGroup(groupId);
+
+        //-----------------------------------------set information related to project
+        project.setProjectName(projectName);
+        project.setGroup(group);
+        //-----------------------------------------set hash
+        String projectHash = getHash(project.getProjectId());
+        hashes.setHash(projectHash);
+        project.setHash(hashes);
+        hashes.setProject(project);
+        em.persist(project);
+        //em.flush();
+        System.out.println("New hash id is"+hashes.getId());
+        addProject(group,project);
+        System.out.println("The number of projects in this group is  " + group.getProjects().size());
+
+        return project;
+
+    }
+
+    private Hash getHashFromBase(int projectId) throws UnsupportedEncodingException, NoSuchAlgorithmException {
```

Comment:
Because getHash() return hash a an initial String and get HashFromBase returns an entity. I was trying to maximally separate functions, so to make the main function look like this: {callmethod1(); callmethod2() .. and so on}.